internal/generate/compile.go

```diff
@@ -98,6 +98,21 @@ func (c *compiler) collectDescriptors() {
 func (c *compiler) addMetadata() {
 	for _, def := range c.defs {
 		switch def := def.(type) {
+		case *dbc.SignalValueTypeDef:
+			signal, ok := c.db.Signal(def.MessageID.ToCAN(), string(def.SignalName))
+			if !ok {
+				c.addWarning(&compileError{def: def, reason: "no declared signal"})
+				continue
+			}
+			err := def.SignalValueType.Validate()
+			if err != nil {
+				c.addWarning(&compileError{def: def, reason: err.Error()})
+			}
+			if def.SignalValueType == dbc.SignalValueTypeFloat32 {
+				signal.IsFloat = true
+			} else {
+				signal.IsFloat = false
+			}
```

Comment:
I think we should only allow values of 0 and 1, so we could actually inline the `def.SignalValueType.Validate()` here.

```suggestion
			switch def.SignalValueType {
			case dbc.SignalValueTypeInt:
				signal.IsFloat = false
			case dbc.SignalValueTypeFloat32:
				signal.IsFloat = true
			default:
				reason := fmt.Sprintf("unsupported signal value type: %v", def.SignalValueType)
				c.addWarning(&compileError{def: def, reason: reason})
			}
```

What do you think about that?

Reply:
Looks good to me, thanks!